PowerEditor/src/menuCmdID.h

```diff
@@ -365,7 +365,9 @@
     #define    IDM_VIEW_SWITCHTO_FUNC_LIST        (IDM_VIEW + 108)
     #define    IDM_VIEW_SWITCHTO_DOCLIST          (IDM_VIEW + 109)
 
-    #define    IDM_VIEW_GOTO_ANOTHER_VIEW        10001
+    #define    IDM_VIEW_CURLINE_FRAMING           (IDM_VIEW + 121)

```

Comment:
This file defines **menu commands** only.

Reply:
@Yaron10,

> This file defines menu commands only.

I was merely replicating this action in the same way how the _Enable current line highlighting_ action is also implemented. AFAIK, _Enable current line highlighting_ is also not a menu option. But we find this in **NppCommands.cpp**:

```C++
		case IDM_VIEW_CURLINE_HILITING:
		{
			NppParameters& nppParams = NppParameters::getInstance();

			COLORREF colour{ nppParams.getCurLineHilitingColour() };
			bool hilite{ nppParams.getSVP()._currentLineHilitingShow };

			_mainEditView.setCurrentLineHiLiting(hilite, colour);
			_subEditView.setCurrentLineHiLiting(hilite, colour);
		}
		break;

		case IDM_VIEW_CURLINE_FRAMING:
		{
			const ScintillaViewParams& svp = NppParameters::getInstance().getSVP();

			bool useFrame{ svp._currentLineUseFrame };
			unsigned char frameWidth{ svp._currentLineFrameWidth };

			_mainEditView.setCurrentLineFraming(useFrame, frameWidth);
			_subEditView.setCurrentLineHiLiting(useFrame, frameWidth);
		}
		break;
```

The benefit from this is that plugins/scripts will gain the ability to invoke these actions via the [NPPM_MENUCOMMAND](https://npp-user-manual.org/docs/plugin-communication/#nppm-menucommand) message.

Reply:
> AFAIK, Enable current line highlighting is also not a menu option. But we find this in NppCommands.cpp:

It was. Then it's been moved into Preferences dialog so it's the historic reason. 

> The benefit from this is that plugins/scripts will gain the ability to invoke these actions via the [NPPM_MENUCOMMAND](https://npp-user-manual.org/docs/plugin-communication/#nppm-menucommand) message.

I'm wondering if it's a good thing - since this option has its GUI part, there will be the incoherence between the state of this option and its GUI.